### PR TITLE
Replace sankey with alluvial plot

### PR DESCRIPTION
**Pull request by mahitha-raj**
- Fixes bug in ggplot2 by replacing the sankey diagram with an alluvial plot
- Mirrors the fix uses in sDAS
- Adds the 'parcats' and 'easyalluvial' packages to NAMESPACE

Note: I used `target_demoData` for the post-QC Aulluvial plot because the text mentioned removing both (i) segments flagged by QC, and (ii) low detection rate. In the original workflow, post-QC `demoData` was used.

### CODE REVIEW

**Review by NicoleEO — CHANGES_REQUESTED** [review]
Line 108, remove the specification for version as GeoMxWorkflows update will be in the yet to be released 3.17. So just do BiocManager::install() and they should all play nice together with the default latest release version installed. I tested to make sure that's the case as well.

File: NAMESPACE [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
-importFrom(data.table, data.table, .SD)
-importFrom(utils, write.table)
-importFrom(outliers, grubbs.test)
-importFrom(BiocGenerics, design)
-importFrom(BiocGenerics, "design<-")
+### - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+### Imports
+import(S4Vectors)
+import(Biobase)
+import(easyalluvial)
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply NAMESPACE:5]: Added https://github.com/Nanostring-Biostats/GeoMxWorkflows/pull/27/commits/6e59118975efcd1d211c0196e9ebd624d5c75bd0

File: NAMESPACE [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
+import(ggplot2)
+import(ggrepel)
+import(ggforce)
+import(cowplot)
+import(scales)
+import(umap)
+import(Rtsne)
+import(parcats)
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply NAMESPACE:13]: Added https://github.com/Nanostring-Biostats/GeoMxWorkflows/pull/27/commits/6e59118975efcd1d211c0196e9ebd624d5c75bd0

File: vignettes/GeomxTools_RNA-NGS_Analysis.Rmd [hunk 3]
```diff
@@ -642,40 +639,31 @@ target_demoData <-
...
+                                 max_variables = 15)
+
+# Add hover option details to plot
+pc <- parcats(p,
+        marginal_histograms = FALSE,
+        data_input = sampleOverview,
+        labelfont = list(size = 12, color = "black"),
+        hoveron="dimension",
+        hoverinfo = "count",
+        offset_marginal_histograms = 0.7)
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply vignettes/GeomxTools_RNA-NGS_Analysis.Rmd]: Removed offset as not required here and changed size of plot. Looks much better when knit!

File: vignettes/GeomxTools_RNA-NGS_Analysis.Rmd [hunk 4]
```diff
@@ -229,42 +229,39 @@ kable(data.frame(PKCs = pkcs, modules = modules))
...
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply vignettes/GeomxTools_RNA-NGS_Analysis.Rmd]: Looks better when knit now: 
image

File: vignettes/GeomxTools_RNA-NGS_Analysis.Rmd [hunk 5]
```diff
@@ -642,40 +639,31 @@ target_demoData <-
...
+                               gsub("normal", "n", sampleOverviewFiltered$`slide name`))
+
+# Create alluvial plot
+p <- easyalluvial::alluvial_wide(sampleOverviewFiltered, 
+                                 max_variables = 15)
+
+# Add hover option details to plot
+pc <- parcats(p,
+        marginal_histograms = FALSE,
+        data_input = sampleOverview,
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply vignettes/GeomxTools_RNA-NGS_Analysis.Rmd]: Thank you for the catch! Fixed [dc9c924](https://github.com/Nanostring-Biostats/GeoMxWorkflows/pull/27/commits/dc9c924bce140a555c457e182575d5a9b9340beb)

File: vignettes/GeomxTools_RNA-NGS_Analysis.Rmd [hunk 6]
```diff
@@ -168,7 +168,7 @@ We first locate the downloaded files:
 ```{r quickstart, message = FALSE, warning = FALSE}
 # Reference the main folder 'file.path' containing the sub-folders with each
 # data file type:
-datadir <- system.file("extdata", "WTA_NGS_Example",
+datadir <- system.file("inst", "extdata", "WTA_NGS_Example",
```
[the remark this answers is not in the document]
Comment (mahitha-raj) [reply vignettes/GeomxTools_RNA-NGS_Analysis.Rmd]: Removed 'inst' and version.

**Comment by mahitha-raj** [comment]
All changes have been made and re-tested the knit. Ready for a follow-up review @NicoleEO .

**Comment by maddygriz** [comment]
@mahitha-raj @NicoleEO I fixed the DE plots. I added the changes here because I couldn't get the vignette to knit on the main branch due to the sankey issue. The alluvial plots look good!

**Comment by NicoleEO** [comment]
Resolves issue #21

**Review by NicoleEO — APPROVED** [review]
Everything in here looks good @mahitha-raj and @maddygriz

**Pull request merged**